src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,222 @@
+            {
+                throw;
+            }
+        }
+
+        private string GenerateProjectFromTemplate(string templateName, string languageName, ITestOutputHelper outputHelper)
+        {
+            var projectPath = GetProjectPath(templateName, languageName);
+            var projectFilePath = GetProjectFilePath(projectPath, languageName);
+
+            var exitCode = NewProject(templateName, projectPath, languageName, outputHelper);
+            Assert.Equal(0, exitCode);
+
+            return projectFilePath;
+        }
+
+        private static async Task AssertProjectLoadsCleanlyAsync(string projectFilePath, string[] ignoredDiagnostics)
+        {
+            using var workspace = CreateMSBuildWorkspace();
+            var project = await workspace.OpenProjectAsync(projectFilePath, cancellationToken: CancellationToken.None);
+
+            Assert.Empty(workspace.Diagnostics);
+
+            var compilation = await project.GetCompilationAsync();
+            Assert.NotNull(compilation);
+
+            // Unnecessary using directives are reported with a severity of Hidden
+            var diagnostics = compilation!.GetDiagnostics()
+                .Where(diagnostic => diagnostic.Severity > DiagnosticSeverity.Hidden && ignoredDiagnostics.Contains(diagnostic.Id) != true);
+
+            Assert.Empty(diagnostics);
+        }
+
+        private void CleanupProject(string templateName, string languageName)
+        {
+            var projectPath = GetProjectPath(templateName, languageName);
+
+            if (Directory.Exists(projectPath))
+            {
+                Directory.Delete(projectPath, true);
+            }
+        }
+
+        private string GetProjectPath(string templateName, string languageName)
+        {
+            var languagePrefix = languageName.Replace("#", "Sharp").Replace(' ', '_').ToLower();
+            var projectName = $"{languagePrefix}_{templateName}_project";
+            return Path.Combine(SolutionDirectory.Path, projectName);
+        }
+
+        private static string GetProjectFilePath(string projectPath, string languageName)
+        {
+            var projectName = Path.GetFileName(projectPath);
+            var projectExtension = languageName switch
+            {
+                LanguageNames.CSharp => "csproj",
+                LanguageNames.VisualBasic => "vbproj",
+                _ => throw new ArgumentOutOfRangeException(nameof(languageName), actualValue: languageName, message: "Only C# and VB.Net project are supported.")
+            };
+            return Path.Combine(projectPath, $"{projectName}.{projectExtension}");
+        }
+
+        public int NewProject(string templateName, string outputPath, string languageName, ITestOutputHelper output)
+        {
+            var language = languageName switch
+            {
+                LanguageNames.CSharp => "C#",
+                LanguageNames.VisualBasic => "VB",
+                LanguageNames.FSharp => "F#",
```

Comment:
Why F# in this list but not the other list?

Reply:
Good call out. Should be removed.

Reply:
Removed